// @adyen/api-library/lib/src/__mocks__/checkout/paymentsDetailsSuccess.js
"use strict";
/*
 *                       ######
 *                       ######
 * ############    ####( ######  #####. ######  ############   ############
 * #############  #####( ######  #####. ######  #############  #############
 *        ######  #####( ######  #####. ######  #####  ######  #####  ######
 * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
 * ###### ######  #####( ######  #####. ######  #####          #####  ######
 * #############  #############  #############  #############  #####  ######
 *  ############   ############  #############   ############  #####  ######
 *                                      ######
 *                               #############
 *                               ############
 * Adyen NodeJS API Library
 * Copyright (c) 2020 Adyen B.V.
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more info.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.paymentDetailsSuccess = void 0;
/* tslint:disable */
exports.paymentDetailsSuccess = JSON.stringify({
    pspReference: "8515232733321252",
    resultCode: "Authorised",
    additionalData: {
        liabilityShift: "true",
        refusalReasonRaw: "AUTHORISED",
    },
});
