// @adyen/api-library/lib/src/services/modification.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const getJsonResponse_1 = __importDefault(require("../helpers/getJsonResponse"));
const service_1 = __importDefault(require("../service"));
const adjustAuthorisation_1 = __importDefault(require("./resource/modification/adjustAuthorisation"));
const cancel_1 = __importDefault(require("./resource/modification/cancel"));
const cancelOrRefund_1 = __importDefault(require("./resource/modification/cancelOrRefund"));
const capture_1 = __importDefault(require("./resource/modification/capture"));
const refund_1 = __importDefault(require("./resource/modification/refund"));
const technicalCancel_1 = __importDefault(require("./resource/modification/technicalCancel"));
const setApplicationInfo_1 = __importDefault(require("../helpers/setApplicationInfo"));
class Modification extends service_1.default {
    constructor(client) {
        super(client);
        this._capture = new capture_1.default(this);
        this._cancelOrRefund = new cancelOrRefund_1.default(this);
        this._cancel = new cancel_1.default(this);
        this._refund = new refund_1.default(this);
        this._adjustAuthorisation = new adjustAuthorisation_1.default(this);
        this._technicalCancel = new technicalCancel_1.default(this);
    }
    capture(captureRequest, requestOptions) {
        return getJsonResponse_1.default(this._capture, setApplicationInfo_1.default(captureRequest), requestOptions);
    }
    cancelOrRefund(cancelOrRefundRequest, requestOptions) {
        return getJsonResponse_1.default(this._cancelOrRefund, setApplicationInfo_1.default(cancelOrRefundRequest), requestOptions);
    }
    refund(refundRequest, requestOptions) {
        return getJsonResponse_1.default(this._refund, setApplicationInfo_1.default(refundRequest), requestOptions);
    }
    cancel(cancelRequest, requestOptions) {
        return getJsonResponse_1.default(this._cancel, setApplicationInfo_1.default(cancelRequest), requestOptions);
    }
    technicalCancel(technicalCancelRequest, requestOptions) {
        return getJsonResponse_1.default(this._technicalCancel, setApplicationInfo_1.default(technicalCancelRequest), requestOptions);
    }
    adjustAuthorisation(adjustAuthorisationRequest, requestOptions) {
        return getJsonResponse_1.default(this._adjustAuthorisation, setApplicationInfo_1.default(adjustAuthorisationRequest), requestOptions);
    }
}
exports.default = Modification;
