// @adyen/api-library/lib/src/services/terminalCloudAPI.js
"use strict";
/*
 *                       ######
 *                       ######
 * ############    ####( ######  #####. ######  ############   ############
 * #############  #####( ######  #####. ######  #############  #############
 *        ######  #####( ######  #####. ######  #####  ######  #####  ######
 * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
 * ###### ######  #####( ######  #####. ######  #####          #####  ######
 * #############  #############  #############  #############  #####  ######
 *  ############   ############  #############   ############  #####  ######
 *                                      ######
 *                               #############
 *                               ############
 * Adyen NodeJS API Library
 * Copyright (c) 2020 Adyen B.V.
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more info.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const apiKeyAuthenticatedService_1 = __importDefault(require("../apiKeyAuthenticatedService"));
const getJsonResponse_1 = __importDefault(require("../helpers/getJsonResponse"));
const async_1 = __importDefault(require("./resource/terminal/cloud/async"));
const sync_1 = __importDefault(require("./resource/terminal/cloud/sync"));
const mergeDeep_1 = __importDefault(require("../utils/mergeDeep"));
const applicationInfo_1 = require("../typings/applicationInfo");
const models_1 = require("../typings/terminal/models");
class TerminalCloudAPI extends apiKeyAuthenticatedService_1.default {
    constructor(client) {
        super(client);
        this.terminalApiAsync = new async_1.default(this);
        this.terminalApiSync = new sync_1.default(this);
    }
    static setApplicationInfo(request) {
        var _a, _b, _c;
        if (request.saleToPOIRequest.paymentRequest) {
            const applicationInfo = new applicationInfo_1.ApplicationInfo();
            const saleToAcquirerData = { applicationInfo };
            const saleData = { saleToAcquirerData };
            const paymentRequest = { saleData };
            const saleToPOIRequest = { paymentRequest };
            const reqWithAppInfo = { saleToPOIRequest };
            mergeDeep_1.default(request, reqWithAppInfo);
            const formattedRequest = models_1.ObjectSerializer.serialize(request, "TerminalApiRequest");
            if ((_c = (_b = (_a = formattedRequest.SaleToPOIRequest) === null || _a === void 0 ? void 0 : _a.PaymentRequest) === null || _b === void 0 ? void 0 : _b.SaleData) === null || _c === void 0 ? void 0 : _c.SaleToAcquirerData) {
                const dataString = JSON.stringify(formattedRequest.SaleToPOIRequest.PaymentRequest.SaleData.SaleToAcquirerData);
                formattedRequest.SaleToPOIRequest.PaymentRequest.SaleData.SaleToAcquirerData = Buffer.from(dataString).toString("base64");
            }
            return formattedRequest;
        }
        return models_1.ObjectSerializer.serialize(request, "TerminalApiRequest");
    }
    async(terminalApiRequest) {
        const request = TerminalCloudAPI.setApplicationInfo(terminalApiRequest);
        return getJsonResponse_1.default(this.terminalApiAsync, request);
    }
    async sync(terminalApiRequest) {
        const request = TerminalCloudAPI.setApplicationInfo(terminalApiRequest);
        const response = await getJsonResponse_1.default(this.terminalApiSync, request);
        return models_1.ObjectSerializer.deserialize(response, "TerminalApiResponse");
    }
}
exports.default = TerminalCloudAPI;
