// @adyen/api-library/lib/src/typings/terminal/barcodeType.js
"use strict";
/*
 *                       ######
 *                       ######
 * ############    ####( ######  #####. ######  ############   ############
 * #############  #####( ######  #####. ######  #############  #############
 *        ######  #####( ######  #####. ######  #####  ######  #####  ######
 * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
 * ###### ######  #####( ######  #####. ######  #####          #####  ######
 * #############  #############  #############  #############  #####  ######
 *  ############   ############  #############   ############  #####  ######
 *                                      ######
 *                               #############
 *                               ############
 * Adyen NodeJS API Library
 * Copyright (c) 2020 Adyen B.V.
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more info.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BarcodeType = void 0;
/**
 * Terminal API
 * Definition of Terminal API Schema
 *
 * The version of the OpenAPI document: 1.0.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var BarcodeType;
(function (BarcodeType) {
    BarcodeType[BarcodeType["Code128"] = "Code128"] = "Code128";
    BarcodeType[BarcodeType["Code25"] = "Code25"] = "Code25";
    BarcodeType[BarcodeType["EAN13"] = "EAN13"] = "EAN13";
    BarcodeType[BarcodeType["EAN8"] = "EAN8"] = "EAN8";
    BarcodeType[BarcodeType["PDF417"] = "PDF417"] = "PDF417";
    BarcodeType[BarcodeType["QRCode"] = "QRCode"] = "QRCode";
    BarcodeType[BarcodeType["UPCA"] = "UPCA"] = "UPCA";
})(BarcodeType = exports.BarcodeType || (exports.BarcodeType = {}));
