// @adyen/api-library/lib/src/typings/terminal/cardReaderInitRequest.js
"use strict";
/*
 *                       ######
 *                       ######
 * ############    ####( ######  #####. ######  ############   ############
 * #############  #####( ######  #####. ######  #############  #############
 *        ######  #####( ######  #####. ######  #####  ######  #####  ######
 * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
 * ###### ######  #####( ######  #####. ######  #####          #####  ######
 * #############  #############  #############  #############  #####  ######
 *  ############   ############  #############   ############  #####  ######
 *                                      ######
 *                               #############
 *                               ############
 * Adyen NodeJS API Library
 * Copyright (c) 2020 Adyen B.V.
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more info.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CardReaderInitRequest = void 0;
class CardReaderInitRequest {
    static getAttributeTypeMap() {
        return CardReaderInitRequest.attributeTypeMap;
    }
}
exports.CardReaderInitRequest = CardReaderInitRequest;
CardReaderInitRequest.discriminator = undefined;
CardReaderInitRequest.attributeTypeMap = [
    {
        "name": "displayOutput",
        "baseName": "DisplayOutput",
        "type": "DisplayOutput"
    },
    {
        "name": "forceEntryMode",
        "baseName": "ForceEntryMode",
        "type": "Array<Array<CardReaderInitRequest.ForceEntryModeEnum>>"
    },
    {
        "name": "leaveCardFlag",
        "baseName": "LeaveCardFlag",
        "type": "boolean"
    },
    {
        "name": "maxWaitingTime",
        "baseName": "MaxWaitingTime",
        "type": "number"
    },
    {
        "name": "warmResetFlag",
        "baseName": "WarmResetFlag",
        "type": "boolean"
    }
];
(function (CardReaderInitRequest) {
    let ForceEntryModeEnum;
    (function (ForceEntryModeEnum) {
        ForceEntryModeEnum[ForceEntryModeEnum["CheckReader"] = "CheckReader"] = "CheckReader";
        ForceEntryModeEnum[ForceEntryModeEnum["Contactless"] = "Contactless"] = "Contactless";
        ForceEntryModeEnum[ForceEntryModeEnum["File"] = "File"] = "File";
        ForceEntryModeEnum[ForceEntryModeEnum["ICC"] = "ICC"] = "ICC";
        ForceEntryModeEnum[ForceEntryModeEnum["Keyed"] = "Keyed"] = "Keyed";
        ForceEntryModeEnum[ForceEntryModeEnum["MagStripe"] = "MagStripe"] = "MagStripe";
        ForceEntryModeEnum[ForceEntryModeEnum["Manual"] = "Manual"] = "Manual";
        ForceEntryModeEnum[ForceEntryModeEnum["RFID"] = "RFID"] = "RFID";
        ForceEntryModeEnum[ForceEntryModeEnum["Scanned"] = "Scanned"] = "Scanned";
        ForceEntryModeEnum[ForceEntryModeEnum["SynchronousICC"] = "SynchronousICC"] = "SynchronousICC";
        ForceEntryModeEnum[ForceEntryModeEnum["Tapped"] = "Tapped"] = "Tapped";
    })(ForceEntryModeEnum = CardReaderInitRequest.ForceEntryModeEnum || (CardReaderInitRequest.ForceEntryModeEnum = {}));
})(CardReaderInitRequest = exports.CardReaderInitRequest || (exports.CardReaderInitRequest = {}));
