// @adyen/api-library/lib/src/typings/terminal/cardData.js
"use strict";
/*
 *                       ######
 *                       ######
 * ############    ####( ######  #####. ######  ############   ############
 * #############  #####( ######  #####. ######  #############  #############
 *        ######  #####( ######  #####. ######  #####  ######  #####  ######
 * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
 * ###### ######  #####( ######  #####. ######  #####          #####  ######
 * #############  #############  #############  #############  #####  ######
 *  ############   ############  #############   ############  #####  ######
 *                                      ######
 *                               #############
 *                               ############
 * Adyen NodeJS API Library
 * Copyright (c) 2020 Adyen B.V.
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more info.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CardData = void 0;
class CardData {
    static getAttributeTypeMap() {
        return CardData.attributeTypeMap;
    }
}
exports.CardData = CardData;
CardData.discriminator = undefined;
CardData.attributeTypeMap = [
    {
        "name": "allowedProduct",
        "baseName": "AllowedProduct",
        "type": "Array<AllowedProduct>"
    },
    {
        "name": "allowedProductCode",
        "baseName": "AllowedProductCode",
        "type": "Array<string>"
    },
    {
        "name": "cardCountryCode",
        "baseName": "CardCountryCode",
        "type": "string"
    },
    {
        "name": "customerOrder",
        "baseName": "CustomerOrder",
        "type": "Array<CustomerOrder>"
    },
    {
        "name": "entryMode",
        "baseName": "EntryMode",
        "type": "Array<CardData.EntryModeEnum>"
    },
    {
        "name": "maskedPAN",
        "baseName": "MaskedPAN",
        "type": "string"
    },
    {
        "name": "paymentAccountRef",
        "baseName": "PaymentAccountRef",
        "type": "string"
    },
    {
        "name": "paymentBrand",
        "baseName": "PaymentBrand",
        "type": "string"
    },
    {
        "name": "paymentToken",
        "baseName": "PaymentToken",
        "type": "PaymentToken"
    },
    {
        "name": "protectedCardData",
        "baseName": "ProtectedCardData",
        "type": "ContentInformation"
    },
    {
        "name": "sensitiveCardData",
        "baseName": "SensitiveCardData",
        "type": "SensitiveCardData"
    }
];
(function (CardData) {
    let EntryModeEnum;
    (function (EntryModeEnum) {
        EntryModeEnum[EntryModeEnum["Contactless"] = "Contactless"] = "Contactless";
        EntryModeEnum[EntryModeEnum["File"] = "File"] = "File";
        EntryModeEnum[EntryModeEnum["ICC"] = "ICC"] = "ICC";
        EntryModeEnum[EntryModeEnum["Keyed"] = "Keyed"] = "Keyed";
        EntryModeEnum[EntryModeEnum["MagStripe"] = "MagStripe"] = "MagStripe";
        EntryModeEnum[EntryModeEnum["Manual"] = "Manual"] = "Manual";
        EntryModeEnum[EntryModeEnum["Mobile"] = "Mobile"] = "Mobile";
        EntryModeEnum[EntryModeEnum["RFID"] = "RFID"] = "RFID";
        EntryModeEnum[EntryModeEnum["Scanned"] = "Scanned"] = "Scanned";
        EntryModeEnum[EntryModeEnum["SynchronousICC"] = "SynchronousICC"] = "SynchronousICC";
        EntryModeEnum[EntryModeEnum["Tapped"] = "Tapped"] = "Tapped";
    })(EntryModeEnum = CardData.EntryModeEnum || (CardData.EntryModeEnum = {}));
})(CardData = exports.CardData || (exports.CardData = {}));
