// @adyen/api-library/lib/src/typings/terminal/cardAcquisitionTransaction.js
"use strict";
/*
 *                       ######
 *                       ######
 * ############    ####( ######  #####. ######  ############   ############
 * #############  #####( ######  #####. ######  #############  #############
 *        ######  #####( ######  #####. ######  #####  ######  #####  ######
 * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
 * ###### ######  #####( ######  #####. ######  #####          #####  ######
 * #############  #############  #############  #############  #####  ######
 *  ############   ############  #############   ############  #####  ######
 *                                      ######
 *                               #############
 *                               ############
 * Adyen NodeJS API Library
 * Copyright (c) 2020 Adyen B.V.
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more info.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CardAcquisitionTransaction = void 0;
/**
 * Terminal API
 * Definition of Terminal API Schema
 *
 * The version of the OpenAPI document: 1.0.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
class CardAcquisitionTransaction {
    static getAttributeTypeMap() {
        return CardAcquisitionTransaction.attributeTypeMap;
    }
}
exports.CardAcquisitionTransaction = CardAcquisitionTransaction;
CardAcquisitionTransaction.discriminator = undefined;
CardAcquisitionTransaction.attributeTypeMap = [
    {
        "name": "allowedLoyaltyBrand",
        "baseName": "AllowedLoyaltyBrand",
        "type": "Array<string>"
    },
    {
        "name": "allowedPaymentBrand",
        "baseName": "AllowedPaymentBrand",
        "type": "Array<string>"
    },
    {
        "name": "cashBackFlag",
        "baseName": "CashBackFlag",
        "type": "boolean"
    },
    {
        "name": "customerLanguage",
        "baseName": "CustomerLanguage",
        "type": "string"
    },
    {
        "name": "forceCustomerSelectionFlag",
        "baseName": "ForceCustomerSelectionFlag",
        "type": "boolean"
    },
    {
        "name": "forceEntryMode",
        "baseName": "ForceEntryMode",
        "type": "Array<Array<CardAcquisitionTransaction.ForceEntryModeEnum>>"
    },
    {
        "name": "loyaltyHandling",
        "baseName": "LoyaltyHandling",
        "type": "CardAcquisitionTransaction.LoyaltyHandlingEnum"
    },
    {
        "name": "paymentType",
        "baseName": "PaymentType",
        "type": "CardAcquisitionTransaction.PaymentTypeEnum"
    },
    {
        "name": "totalAmount",
        "baseName": "TotalAmount",
        "type": "number"
    }
];
(function (CardAcquisitionTransaction) {
    let ForceEntryModeEnum;
    (function (ForceEntryModeEnum) {
        ForceEntryModeEnum[ForceEntryModeEnum["CheckReader"] = "CheckReader"] = "CheckReader";
        ForceEntryModeEnum[ForceEntryModeEnum["Contactless"] = "Contactless"] = "Contactless";
        ForceEntryModeEnum[ForceEntryModeEnum["File"] = "File"] = "File";
        ForceEntryModeEnum[ForceEntryModeEnum["ICC"] = "ICC"] = "ICC";
        ForceEntryModeEnum[ForceEntryModeEnum["Keyed"] = "Keyed"] = "Keyed";
        ForceEntryModeEnum[ForceEntryModeEnum["MagStripe"] = "MagStripe"] = "MagStripe";
        ForceEntryModeEnum[ForceEntryModeEnum["Manual"] = "Manual"] = "Manual";
        ForceEntryModeEnum[ForceEntryModeEnum["RFID"] = "RFID"] = "RFID";
        ForceEntryModeEnum[ForceEntryModeEnum["Scanned"] = "Scanned"] = "Scanned";
        ForceEntryModeEnum[ForceEntryModeEnum["SynchronousICC"] = "SynchronousICC"] = "SynchronousICC";
        ForceEntryModeEnum[ForceEntryModeEnum["Tapped"] = "Tapped"] = "Tapped";
    })(ForceEntryModeEnum = CardAcquisitionTransaction.ForceEntryModeEnum || (CardAcquisitionTransaction.ForceEntryModeEnum = {}));
    let LoyaltyHandlingEnum;
    (function (LoyaltyHandlingEnum) {
        LoyaltyHandlingEnum[LoyaltyHandlingEnum["Allowed"] = "Allowed"] = "Allowed";
        LoyaltyHandlingEnum[LoyaltyHandlingEnum["Forbidden"] = "Forbidden"] = "Forbidden";
        LoyaltyHandlingEnum[LoyaltyHandlingEnum["Processed"] = "Processed"] = "Processed";
        LoyaltyHandlingEnum[LoyaltyHandlingEnum["Proposed"] = "Proposed"] = "Proposed";
        LoyaltyHandlingEnum[LoyaltyHandlingEnum["Required"] = "Required"] = "Required";
    })(LoyaltyHandlingEnum = CardAcquisitionTransaction.LoyaltyHandlingEnum || (CardAcquisitionTransaction.LoyaltyHandlingEnum = {}));
    let PaymentTypeEnum;
    (function (PaymentTypeEnum) {
        PaymentTypeEnum[PaymentTypeEnum["CashAdvance"] = "CashAdvance"] = "CashAdvance";
        PaymentTypeEnum[PaymentTypeEnum["CashDeposit"] = "CashDeposit"] = "CashDeposit";
        PaymentTypeEnum[PaymentTypeEnum["Completion"] = "Completion"] = "Completion";
        PaymentTypeEnum[PaymentTypeEnum["FirstReservation"] = "FirstReservation"] = "FirstReservation";
        PaymentTypeEnum[PaymentTypeEnum["Instalment"] = "Instalment"] = "Instalment";
        PaymentTypeEnum[PaymentTypeEnum["IssuerInstalment"] = "IssuerInstalment"] = "IssuerInstalment";
        PaymentTypeEnum[PaymentTypeEnum["Normal"] = "Normal"] = "Normal";
        PaymentTypeEnum[PaymentTypeEnum["OneTimeReservation"] = "OneTimeReservation"] = "OneTimeReservation";
        PaymentTypeEnum[PaymentTypeEnum["PaidOut"] = "PaidOut"] = "PaidOut";
        PaymentTypeEnum[PaymentTypeEnum["Recurring"] = "Recurring"] = "Recurring";
        PaymentTypeEnum[PaymentTypeEnum["Refund"] = "Refund"] = "Refund";
        PaymentTypeEnum[PaymentTypeEnum["UpdateReservation"] = "UpdateReservation"] = "UpdateReservation";
    })(PaymentTypeEnum = CardAcquisitionTransaction.PaymentTypeEnum || (CardAcquisitionTransaction.PaymentTypeEnum = {}));
})(CardAcquisitionTransaction = exports.CardAcquisitionTransaction || (exports.CardAcquisitionTransaction = {}));
