// @adyen/api-library/lib/src/typings/terminal/menuEntry.js
"use strict";
/*
 *                       ######
 *                       ######
 * ############    ####( ######  #####. ######  ############   ############
 * #############  #####( ######  #####. ######  #############  #############
 *        ######  #####( ######  #####. ######  #####  ######  #####  ######
 * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
 * ###### ######  #####( ######  #####. ######  #####          #####  ######
 * #############  #############  #############  #############  #####  ######
 *  ############   ############  #############   ############  #####  ######
 *                                      ######
 *                               #############
 *                               ############
 * Adyen NodeJS API Library
 * Copyright (c) 2020 Adyen B.V.
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more info.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MenuEntry = void 0;
class MenuEntry {
    static getAttributeTypeMap() {
        return MenuEntry.attributeTypeMap;
    }
}
exports.MenuEntry = MenuEntry;
MenuEntry.discriminator = undefined;
MenuEntry.attributeTypeMap = [
    {
        "name": "defaultSelectedFlag",
        "baseName": "DefaultSelectedFlag",
        "type": "boolean"
    },
    {
        "name": "menuEntryTag",
        "baseName": "MenuEntryTag",
        "type": "MenuEntry.MenuEntryTagEnum"
    },
    {
        "name": "outputFormat",
        "baseName": "OutputFormat",
        "type": "OutputFormatType"
    },
    {
        "name": "outputText",
        "baseName": "OutputText",
        "type": "Array<OutputText>"
    },
    {
        "name": "outputXHTML",
        "baseName": "OutputXHTML",
        "type": "AnyType"
    },
    {
        "name": "predefinedContent",
        "baseName": "PredefinedContent",
        "type": "PredefinedContent"
    }
];
(function (MenuEntry) {
    let MenuEntryTagEnum;
    (function (MenuEntryTagEnum) {
        MenuEntryTagEnum[MenuEntryTagEnum["NonSelectable"] = "NonSelectable"] = "NonSelectable";
        MenuEntryTagEnum[MenuEntryTagEnum["NonSelectableSubMenu"] = "NonSelectableSubMenu"] = "NonSelectableSubMenu";
        MenuEntryTagEnum[MenuEntryTagEnum["Selectable"] = "Selectable"] = "Selectable";
        MenuEntryTagEnum[MenuEntryTagEnum["SubMenu"] = "SubMenu"] = "SubMenu";
    })(MenuEntryTagEnum = MenuEntry.MenuEntryTagEnum || (MenuEntry.MenuEntryTagEnum = {}));
})(MenuEntry = exports.MenuEntry || (exports.MenuEntry = {}));
