// @adyen/api-library/lib/src/typings/terminal/outputBarcode.js
"use strict";
/*
 *                       ######
 *                       ######
 * ############    ####( ######  #####. ######  ############   ############
 * #############  #####( ######  #####. ######  #############  #############
 *        ######  #####( ######  #####. ######  #####  ######  #####  ######
 * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
 * ###### ######  #####( ######  #####. ######  #####          #####  ######
 * #############  #############  #############  #############  #####  ######
 *  ############   ############  #############   ############  #####  ######
 *                                      ######
 *                               #############
 *                               ############
 * Adyen NodeJS API Library
 * Copyright (c) 2020 Adyen B.V.
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more info.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OutputBarcode = void 0;
/**
 * Terminal API
 * Definition of Terminal API Schema
 *
 * The version of the OpenAPI document: 1.0.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
class OutputBarcode {
    static getAttributeTypeMap() {
        return OutputBarcode.attributeTypeMap;
    }
}
exports.OutputBarcode = OutputBarcode;
OutputBarcode.discriminator = undefined;
OutputBarcode.attributeTypeMap = [
    {
        "name": "barcodeType",
        "baseName": "BarcodeType",
        "type": "OutputBarcode.BarcodeTypeEnum"
    },
    {
        "name": "barcodeValue",
        "baseName": "BarcodeValue",
        "type": "string"
    }
];
(function (OutputBarcode) {
    let BarcodeTypeEnum;
    (function (BarcodeTypeEnum) {
        BarcodeTypeEnum[BarcodeTypeEnum["Code128"] = "Code128"] = "Code128";
        BarcodeTypeEnum[BarcodeTypeEnum["Code25"] = "Code25"] = "Code25";
        BarcodeTypeEnum[BarcodeTypeEnum["EAN13"] = "EAN13"] = "EAN13";
        BarcodeTypeEnum[BarcodeTypeEnum["EAN8"] = "EAN8"] = "EAN8";
        BarcodeTypeEnum[BarcodeTypeEnum["PDF417"] = "PDF417"] = "PDF417";
        BarcodeTypeEnum[BarcodeTypeEnum["QRCode"] = "QRCode"] = "QRCode";
        BarcodeTypeEnum[BarcodeTypeEnum["UPCA"] = "UPCA"] = "UPCA";
    })(BarcodeTypeEnum = OutputBarcode.BarcodeTypeEnum || (OutputBarcode.BarcodeTypeEnum = {}));
})(OutputBarcode = exports.OutputBarcode || (exports.OutputBarcode = {}));
