// @adyen/api-library/lib/src/typings/terminal/paymentResult.js
"use strict";
/*
 *                       ######
 *                       ######
 * ############    ####( ######  #####. ######  ############   ############
 * #############  #####( ######  #####. ######  #############  #############
 *        ######  #####( ######  #####. ######  #####  ######  #####  ######
 * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
 * ###### ######  #####( ######  #####. ######  #####          #####  ######
 * #############  #############  #############  #############  #####  ######
 *  ############   ############  #############   ############  #####  ######
 *                                      ######
 *                               #############
 *                               ############
 * Adyen NodeJS API Library
 * Copyright (c) 2020 Adyen B.V.
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more info.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentResult = void 0;
class PaymentResult {
    static getAttributeTypeMap() {
        return PaymentResult.attributeTypeMap;
    }
}
exports.PaymentResult = PaymentResult;
PaymentResult.discriminator = undefined;
PaymentResult.attributeTypeMap = [
    {
        "name": "amountsResp",
        "baseName": "AmountsResp",
        "type": "AmountsResp"
    },
    {
        "name": "authenticationMethod",
        "baseName": "AuthenticationMethod",
        "type": "Array<PaymentResult.AuthenticationMethodEnum>"
    },
    {
        "name": "capturedSignature",
        "baseName": "CapturedSignature",
        "type": "CapturedSignature"
    },
    {
        "name": "currencyConversion",
        "baseName": "CurrencyConversion",
        "type": "Array<CurrencyConversion>"
    },
    {
        "name": "customerLanguage",
        "baseName": "CustomerLanguage",
        "type": "string"
    },
    {
        "name": "instalmentType",
        "baseName": "InstalmentType",
        "type": "Instalment"
    },
    {
        "name": "merchantOverrideFlag",
        "baseName": "MerchantOverrideFlag",
        "type": "boolean"
    },
    {
        "name": "onlineFlag",
        "baseName": "OnlineFlag",
        "type": "boolean"
    },
    {
        "name": "paymentAcquirerData",
        "baseName": "PaymentAcquirerData",
        "type": "PaymentAcquirerData"
    },
    {
        "name": "paymentInstrumentData",
        "baseName": "PaymentInstrumentData",
        "type": "PaymentInstrumentData"
    },
    {
        "name": "paymentType",
        "baseName": "PaymentType",
        "type": "PaymentResult.PaymentTypeEnum"
    },
    {
        "name": "protectedSignature",
        "baseName": "ProtectedSignature",
        "type": "ContentInformation"
    },
    {
        "name": "validityDate",
        "baseName": "ValidityDate",
        "type": "string"
    }
];
(function (PaymentResult) {
    let AuthenticationMethodEnum;
    (function (AuthenticationMethodEnum) {
        AuthenticationMethodEnum[AuthenticationMethodEnum["Bypass"] = "Bypass"] = "Bypass";
        AuthenticationMethodEnum[AuthenticationMethodEnum["ManualVerification"] = "ManualVerification"] = "ManualVerification";
        AuthenticationMethodEnum[AuthenticationMethodEnum["MerchantAuthentication"] = "MerchantAuthentication"] = "MerchantAuthentication";
        AuthenticationMethodEnum[AuthenticationMethodEnum["OfflinePIN"] = "OfflinePIN"] = "OfflinePIN";
        AuthenticationMethodEnum[AuthenticationMethodEnum["OnlinePIN"] = "OnlinePIN"] = "OnlinePIN";
        AuthenticationMethodEnum[AuthenticationMethodEnum["PaperSignature"] = "PaperSignature"] = "PaperSignature";
        AuthenticationMethodEnum[AuthenticationMethodEnum["SecureCertificate"] = "SecureCertificate"] = "SecureCertificate";
        AuthenticationMethodEnum[AuthenticationMethodEnum["SecureNoCertificate"] = "SecureNoCertificate"] = "SecureNoCertificate";
        AuthenticationMethodEnum[AuthenticationMethodEnum["SecuredChannel"] = "SecuredChannel"] = "SecuredChannel";
        AuthenticationMethodEnum[AuthenticationMethodEnum["SignatureCapture"] = "SignatureCapture"] = "SignatureCapture";
        AuthenticationMethodEnum[AuthenticationMethodEnum["UnknownMethod"] = "UnknownMethod"] = "UnknownMethod";
    })(AuthenticationMethodEnum = PaymentResult.AuthenticationMethodEnum || (PaymentResult.AuthenticationMethodEnum = {}));
    let PaymentTypeEnum;
    (function (PaymentTypeEnum) {
        PaymentTypeEnum[PaymentTypeEnum["CashAdvance"] = "CashAdvance"] = "CashAdvance";
        PaymentTypeEnum[PaymentTypeEnum["CashDeposit"] = "CashDeposit"] = "CashDeposit";
        PaymentTypeEnum[PaymentTypeEnum["Completion"] = "Completion"] = "Completion";
        PaymentTypeEnum[PaymentTypeEnum["FirstReservation"] = "FirstReservation"] = "FirstReservation";
        PaymentTypeEnum[PaymentTypeEnum["Instalment"] = "Instalment"] = "Instalment";
        PaymentTypeEnum[PaymentTypeEnum["IssuerInstalment"] = "IssuerInstalment"] = "IssuerInstalment";
        PaymentTypeEnum[PaymentTypeEnum["Normal"] = "Normal"] = "Normal";
        PaymentTypeEnum[PaymentTypeEnum["OneTimeReservation"] = "OneTimeReservation"] = "OneTimeReservation";
        PaymentTypeEnum[PaymentTypeEnum["PaidOut"] = "PaidOut"] = "PaidOut";
        PaymentTypeEnum[PaymentTypeEnum["Recurring"] = "Recurring"] = "Recurring";
        PaymentTypeEnum[PaymentTypeEnum["Refund"] = "Refund"] = "Refund";
        PaymentTypeEnum[PaymentTypeEnum["UpdateReservation"] = "UpdateReservation"] = "UpdateReservation";
    })(PaymentTypeEnum = PaymentResult.PaymentTypeEnum || (PaymentResult.PaymentTypeEnum = {}));
})(PaymentResult = exports.PaymentResult || (exports.PaymentResult = {}));
