// @adyen/api-library/lib/src/typings/terminal/pOIStatus.js
"use strict";
/*
 *                       ######
 *                       ######
 * ############    ####( ######  #####. ######  ############   ############
 * #############  #####( ######  #####. ######  #############  #############
 *        ######  #####( ######  #####. ######  #####  ######  #####  ######
 * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
 * ###### ######  #####( ######  #####. ######  #####          #####  ######
 * #############  #############  #############  #############  #####  ######
 *  ############   ############  #############   ############  #####  ######
 *                                      ######
 *                               #############
 *                               ############
 * Adyen NodeJS API Library
 * Copyright (c) 2020 Adyen B.V.
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more info.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.POIStatus = void 0;
class POIStatus {
    static getAttributeTypeMap() {
        return POIStatus.attributeTypeMap;
    }
}
exports.POIStatus = POIStatus;
POIStatus.discriminator = undefined;
POIStatus.attributeTypeMap = [
    {
        "name": "cardReaderOkFlag",
        "baseName": "CardReaderOkFlag",
        "type": "boolean"
    },
    {
        "name": "cashHandlingDevice",
        "baseName": "CashHandlingDevice",
        "type": "Array<CashHandlingDevice>"
    },
    {
        "name": "communicationOkFlag",
        "baseName": "CommunicationOkFlag",
        "type": "boolean"
    },
    {
        "name": "fraudPreventionFlag",
        "baseName": "FraudPreventionFlag",
        "type": "boolean"
    },
    {
        "name": "globalStatus",
        "baseName": "GlobalStatus",
        "type": "GlobalStatusType"
    },
    {
        "name": "pEDOKFlag",
        "baseName": "PEDOKFlag",
        "type": "boolean"
    },
    {
        "name": "printerStatus",
        "baseName": "PrinterStatus",
        "type": "POIStatus.PrinterStatusEnum"
    },
    {
        "name": "securityOkFlag",
        "baseName": "SecurityOkFlag",
        "type": "boolean"
    }
];
(function (POIStatus) {
    let PrinterStatusEnum;
    (function (PrinterStatusEnum) {
        PrinterStatusEnum[PrinterStatusEnum["NoPaper"] = "NoPaper"] = "NoPaper";
        PrinterStatusEnum[PrinterStatusEnum["OK"] = "OK"] = "OK";
        PrinterStatusEnum[PrinterStatusEnum["OutOfOrder"] = "OutOfOrder"] = "OutOfOrder";
        PrinterStatusEnum[PrinterStatusEnum["PaperJam"] = "PaperJam"] = "PaperJam";
        PrinterStatusEnum[PrinterStatusEnum["PaperLow"] = "PaperLow"] = "PaperLow";
    })(PrinterStatusEnum = POIStatus.PrinterStatusEnum || (POIStatus.PrinterStatusEnum = {}));
})(POIStatus = exports.POIStatus || (exports.POIStatus = {}));
