// @adyen/api-library/lib/src/typings/terminal/soundRequest.js
"use strict";
/*
 *                       ######
 *                       ######
 * ############    ####( ######  #####. ######  ############   ############
 * #############  #####( ######  #####. ######  #############  #############
 *        ######  #####( ######  #####. ######  #####  ######  #####  ######
 * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
 * ###### ######  #####( ######  #####. ######  #####          #####  ######
 * #############  #############  #############  #############  #####  ######
 *  ############   ############  #############   ############  #####  ######
 *                                      ######
 *                               #############
 *                               ############
 * Adyen NodeJS API Library
 * Copyright (c) 2020 Adyen B.V.
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more info.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SoundRequest = void 0;
class SoundRequest {
    static getAttributeTypeMap() {
        return SoundRequest.attributeTypeMap;
    }
}
exports.SoundRequest = SoundRequest;
SoundRequest.discriminator = undefined;
SoundRequest.attributeTypeMap = [
    {
        "name": "responseMode",
        "baseName": "ResponseMode",
        "type": "SoundRequest.ResponseModeEnum"
    },
    {
        "name": "soundAction",
        "baseName": "SoundAction",
        "type": "SoundActionType"
    },
    {
        "name": "soundContent",
        "baseName": "SoundContent",
        "type": "SoundContent"
    },
    {
        "name": "soundVolume",
        "baseName": "SoundVolume",
        "type": "number"
    }
];
(function (SoundRequest) {
    let ResponseModeEnum;
    (function (ResponseModeEnum) {
        ResponseModeEnum[ResponseModeEnum["Immediate"] = "Immediate"] = "Immediate";
        ResponseModeEnum[ResponseModeEnum["NotRequired"] = "NotRequired"] = "NotRequired";
        ResponseModeEnum[ResponseModeEnum["PrintEnd"] = "PrintEnd"] = "PrintEnd";
        ResponseModeEnum[ResponseModeEnum["SoundEnd"] = "SoundEnd"] = "SoundEnd";
    })(ResponseModeEnum = SoundRequest.ResponseModeEnum || (SoundRequest.ResponseModeEnum = {}));
})(SoundRequest = exports.SoundRequest || (exports.SoundRequest = {}));
