// @adyen/api-library/lib/src/typings/terminal/transactionConditions.js
"use strict";
/*
 *                       ######
 *                       ######
 * ############    ####( ######  #####. ######  ############   ############
 * #############  #####( ######  #####. ######  #############  #############
 *        ######  #####( ######  #####. ######  #####  ######  #####  ######
 * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
 * ###### ######  #####( ######  #####. ######  #####          #####  ######
 * #############  #############  #############  #############  #####  ######
 *  ############   ############  #############   ############  #####  ######
 *                                      ######
 *                               #############
 *                               ############
 * Adyen NodeJS API Library
 * Copyright (c) 2020 Adyen B.V.
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more info.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TransactionConditions = void 0;
/**
 * Terminal API
 * Definition of Terminal API Schema
 *
 * The version of the OpenAPI document: 1.0.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
class TransactionConditions {
    static getAttributeTypeMap() {
        return TransactionConditions.attributeTypeMap;
    }
}
exports.TransactionConditions = TransactionConditions;
TransactionConditions.discriminator = undefined;
TransactionConditions.attributeTypeMap = [
    {
        "name": "acquirerID",
        "baseName": "AcquirerID",
        "type": "Array<string>"
    },
    {
        "name": "allowedLoyaltyBrand",
        "baseName": "AllowedLoyaltyBrand",
        "type": "Array<string>"
    },
    {
        "name": "allowedPaymentBrand",
        "baseName": "AllowedPaymentBrand",
        "type": "Array<string>"
    },
    {
        "name": "customerLanguage",
        "baseName": "CustomerLanguage",
        "type": "string"
    },
    {
        "name": "debitPreferredFlag",
        "baseName": "DebitPreferredFlag",
        "type": "boolean"
    },
    {
        "name": "forceEntryMode",
        "baseName": "ForceEntryMode",
        "type": "Array<Array<TransactionConditions.ForceEntryModeEnum>>"
    },
    {
        "name": "forceOnlineFlag",
        "baseName": "ForceOnlineFlag",
        "type": "boolean"
    },
    {
        "name": "loyaltyHandling",
        "baseName": "LoyaltyHandling",
        "type": "TransactionConditions.LoyaltyHandlingEnum"
    },
    {
        "name": "merchantCategoryCode",
        "baseName": "MerchantCategoryCode",
        "type": "string"
    }
];
(function (TransactionConditions) {
    let ForceEntryModeEnum;
    (function (ForceEntryModeEnum) {
        ForceEntryModeEnum[ForceEntryModeEnum["CheckReader"] = "CheckReader"] = "CheckReader";
        ForceEntryModeEnum[ForceEntryModeEnum["Contactless"] = "Contactless"] = "Contactless";
        ForceEntryModeEnum[ForceEntryModeEnum["File"] = "File"] = "File";
        ForceEntryModeEnum[ForceEntryModeEnum["ICC"] = "ICC"] = "ICC";
        ForceEntryModeEnum[ForceEntryModeEnum["Keyed"] = "Keyed"] = "Keyed";
        ForceEntryModeEnum[ForceEntryModeEnum["MagStripe"] = "MagStripe"] = "MagStripe";
        ForceEntryModeEnum[ForceEntryModeEnum["Manual"] = "Manual"] = "Manual";
        ForceEntryModeEnum[ForceEntryModeEnum["RFID"] = "RFID"] = "RFID";
        ForceEntryModeEnum[ForceEntryModeEnum["Scanned"] = "Scanned"] = "Scanned";
        ForceEntryModeEnum[ForceEntryModeEnum["SynchronousICC"] = "SynchronousICC"] = "SynchronousICC";
        ForceEntryModeEnum[ForceEntryModeEnum["Tapped"] = "Tapped"] = "Tapped";
    })(ForceEntryModeEnum = TransactionConditions.ForceEntryModeEnum || (TransactionConditions.ForceEntryModeEnum = {}));
    let LoyaltyHandlingEnum;
    (function (LoyaltyHandlingEnum) {
        LoyaltyHandlingEnum[LoyaltyHandlingEnum["Allowed"] = "Allowed"] = "Allowed";
        LoyaltyHandlingEnum[LoyaltyHandlingEnum["Forbidden"] = "Forbidden"] = "Forbidden";
        LoyaltyHandlingEnum[LoyaltyHandlingEnum["Processed"] = "Processed"] = "Processed";
        LoyaltyHandlingEnum[LoyaltyHandlingEnum["Proposed"] = "Proposed"] = "Proposed";
        LoyaltyHandlingEnum[LoyaltyHandlingEnum["Required"] = "Required"] = "Required";
    })(LoyaltyHandlingEnum = TransactionConditions.LoyaltyHandlingEnum || (TransactionConditions.LoyaltyHandlingEnum = {}));
})(TransactionConditions = exports.TransactionConditions || (exports.TransactionConditions = {}));
