// @adyen/api-library/lib/src/__tests__/platforms.spec.js
"use strict";
/*
 *                       ######
 *                       ######
 * ############    ####( ######  #####. ######  ############   ############
 * #############  #####( ######  #####. ######  #############  #############
 *        ######  #####( ######  #####. ######  #####  ######  #####  ######
 * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
 * ###### ######  #####( ######  #####. ######  #####          #####  ######
 * #############  #############  #############  #############  #####  ######
 *  ############   ############  #############   ############  #####  ######
 *                                      ######
 *                               #############
 *                               ############
 * Adyen NodeJS API Library
 * Copyright (c) 2020 Adyen B.V.
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more info.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const nock_1 = __importDefault(require("nock"));
const ts_auto_mock_1 = require("ts-auto-mock");
const base_1 = require("../__mocks__/base");
const documentContent_1 = require("../__mocks__/platforms/documentContent");
const index_1 = require("../index");
let client;
let platforms;
let scope;
let accountHolder;
let account;
let accountHolderToSuspend;
let accountToClose;
let accountHolderToUnSuspend;
let accountHolderToClose;
let notificationConfigurationToRetrieve;
const generateRandomCode = () => Math.floor(Math.random() * Date.now()).toString();
const accountHolderDetails = {
    email: "random_email@example.com",
    fullPhoneNumber: "312030291928",
    webAddress: "http://example.com",
    individualDetails: {
        name: {
            firstName: "John",
            gender: "MALE",
            lastName: "Smith"
        }
    },
    address: {
        country: "NL"
    },
};
const notificationConfigurationDetails = {
    active: true,
    notifyURL: "https://www.adyen.com/notification-handler",
    eventConfigs: [
        {
            eventType: "ACCOUNT_HOLDER_VERIFICATION",
            includeMode: "INCLUDE"
        }
    ],
    sslProtocol: "SSL"
};
const assertError = (e) => {
    var _a, _b;
    if (((_a = e.responseBody) === null || _a === void 0 ? void 0 : _a.includes("Account code does not exist or invalid")) || ((_b = e.responseBody) === null || _b === void 0 ? void 0 : _b.includes("Failed to retrieve account holder"))) {
        return;
    }
    fail(e);
};
beforeEach(() => {
    if (!nock_1.default.isActive()) {
        nock_1.default.activate();
    }
    client = base_1.createBasicAuthClient();
    client.config.password = process.env.ADYEN_MARKETPLACE_PASSWORD;
    client.config.username = process.env.ADYEN_MARKETPLACE_USER;
    scope = nock_1.default(client.config.marketPayEndpoint);
    platforms = new index_1.Platforms(client);
});
afterEach(() => {
    nock_1.default.cleanAll();
});
describe("Platforms Test", function () {
    describe("Account", function () {
        describe("Accounts", function () {
            const cases = [
                ["closeAccount", ts_auto_mock_1.createMock(), ts_auto_mock_1.createMock()],
                ["updateAccount", ts_auto_mock_1.createMock(), ts_auto_mock_1.createMock()],
                ["createAccount", ts_auto_mock_1.createMock(), ts_auto_mock_1.createMock()],
                ["uploadDocument", ts_auto_mock_1.createMock(), ts_auto_mock_1.createMock()],
                ["getUploadedDocuments", ts_auto_mock_1.createMock(), ts_auto_mock_1.createMock()],
                ["deleteBankAccounts", ts_auto_mock_1.createMock(), ts_auto_mock_1.createMock()],
                ["deletePayoutMethods", ts_auto_mock_1.createMock(), ts_auto_mock_1.createMock()],
                ["deleteShareholders", ts_auto_mock_1.createMock(), ts_auto_mock_1.createMock()],
                ["checkAccountHolder", ts_auto_mock_1.createMock(), ts_auto_mock_1.createMock()],
                ["createAccountHolder", ts_auto_mock_1.createMock(), ts_auto_mock_1.createMock()],
                ["getAccountHolder", ts_auto_mock_1.createMock(), ts_auto_mock_1.createMock()],
                ["updateAccountHolder", ts_auto_mock_1.createMock(), ts_auto_mock_1.createMock()],
                ["updateAccountHolderState", ts_auto_mock_1.createMock(), ts_auto_mock_1.createMock()],
                ["suspendAccountHolder", ts_auto_mock_1.createMock(), ts_auto_mock_1.createMock()],
                ["unSuspendAccountHolder", ts_auto_mock_1.createMock(), ts_auto_mock_1.createMock()],
                ["closeAccountHolder", ts_auto_mock_1.createMock(), ts_auto_mock_1.createMock()],
            ];
            test.each(cases)("should %p", async (...args) => {
                const service = platforms.Account;
                scope.post(`/Account/${index_1.Client.MARKETPAY_ACCOUNT_API_VERSION}//${args[0]}`).reply(200, args[2]);
                const result = await service[args[0]](args[1]);
                expect(result).toMatchObject(args[2]);
            });
        });
    });
    describe("Fund", function () {
        const cases = [
            ["accountHolderBalance", ts_auto_mock_1.createMock(), ts_auto_mock_1.createMock()],
            ["accountHolderTransactionList", ts_auto_mock_1.createMock(), ts_auto_mock_1.createMock()],
            ["payoutAccountHolder", ts_auto_mock_1.createMock(), ts_auto_mock_1.createMock()],
            ["transferFunds", ts_auto_mock_1.createMock(), ts_auto_mock_1.createMock()],
            ["refundFundsTransfer", ts_auto_mock_1.createMock(), ts_auto_mock_1.createMock()],
            ["setupBeneficiary", ts_auto_mock_1.createMock(), ts_auto_mock_1.createMock()],
            ["refundNotPaidOutTransfers", ts_auto_mock_1.createMock(), ts_auto_mock_1.createMock()],
        ];
        test.each(cases)("should %p", async (...args) => {
            const fund = platforms.Fund;
            scope.post(`/Fund/${index_1.Client.MARKETPAY_FUND_API_VERSION}//${args[0]}`).reply(200, args[2]);
            const result = await fund[args[0]](args[1]);
            expect(result).toMatchObject(args[2]);
        });
    });
    describe("Notification Configuration", function () {
        const cases = [
            ["createNotificationConfiguration", ts_auto_mock_1.createMock(), ts_auto_mock_1.createMock()],
            ["getNotificationConfiguration", ts_auto_mock_1.createMock(), ts_auto_mock_1.createMock()],
            ["getNotificationConfigurationList", {}, ts_auto_mock_1.createMock()],
            ["testNotificationConfiguration", ts_auto_mock_1.createMock(), ts_auto_mock_1.createMock()],
            ["updateNotificationConfiguration", ts_auto_mock_1.createMock(), ts_auto_mock_1.createMock()],
            ["deleteNotificationConfigurations", ts_auto_mock_1.createMock(), ts_auto_mock_1.createMock()],
        ];
        test.each(cases)("should %p", async (...args) => {
            const notificationConfiguration = platforms.NotificationConfiguration;
            scope.post(`/Notification/${index_1.Client.MARKETPAY_NOTIFICATION_CONFIGURATION_API_VERSION}//${args[0]}`).reply(200, args[2]);
            const result = await notificationConfiguration[args[0]](args[1]);
            expect(result).toMatchObject(args[2]);
        });
    });
    describe("Hop", function () {
        const cases = [
            ["getOnboardingUrl", ts_auto_mock_1.createMock(), ts_auto_mock_1.createMock()]
        ];
        test.each(cases)("should %p", async (...args) => {
            const hostedOnboardingPage = platforms.HostedOnboardingPage;
            scope.post(`/Hop/${index_1.Client.MARKETPAY_HOP_API_VERSION}//${args[0]}`).reply(200, args[2]);
            const result = await hostedOnboardingPage[args[0]](args[1]);
            expect(result).toMatchObject(args[2]);
        });
    });
});
describe.skip("Platforms Test E2E", function () {
    beforeAll(async (done) => {
        accountHolder = await platforms.Account.createAccountHolder({
            accountHolderCode: generateRandomCode(),
            accountHolderDetails,
            legalEntity: "Individual",
        });
        account = await platforms.Account.createAccount({
            accountHolderCode: generateRandomCode(),
            description: "This is a new account",
            metadata: { meta: "data" },
            payoutSchedule: "WEEKLY"
        });
        accountHolderToSuspend = await platforms.Account.createAccountHolder({
            accountHolderCode: generateRandomCode(),
            accountHolderDetails,
            legalEntity: "Individual"
        });
        accountToClose = await platforms.Account.createAccount({
            accountHolderCode: generateRandomCode(),
            description: "This is a new account",
            metadata: { meta: "data" },
            payoutSchedule: "WEEKLY"
        });
        accountHolderToUnSuspend = await platforms.Account.createAccountHolder({
            accountHolderCode: generateRandomCode(),
            accountHolderDetails,
            legalEntity: "Individual"
        });
        await platforms.Account.suspendAccountHolder({ accountHolderCode: accountHolderToUnSuspend.accountHolderCode });
        accountHolderToClose = await platforms.Account.createAccountHolder({
            accountHolderCode: generateRandomCode(),
            accountHolderDetails,
            legalEntity: "Individual"
        });
        notificationConfigurationToRetrieve = await platforms.NotificationConfiguration.createNotificationConfiguration({
            configurationDetails: {
                ...notificationConfigurationDetails,
                description: `${generateRandomCode()}`
            }
        });
        done();
    });
    describe("Account", function () {
        describe("Accounts E2E", function () {
            it("should create account holder", async function () {
                nock_1.default.restore();
                try {
                    expect(accountHolder.pspReference).toBeDefined();
                }
                catch (e) {
                    assertError(e);
                }
            });
            it("should get account holder", async function () {
                nock_1.default.restore();
                try {
                    const result = await platforms.Account.getAccountHolder({
                        accountHolderCode: accountHolder.accountHolderCode,
                    });
                    expect(result.accountHolderDetails.email).toEqual("random_email@example.com");
                }
                catch (e) {
                    assertError(e);
                }
            });
            it("should update account holder", async function () {
                var _a;
                nock_1.default.restore();
                try {
                    const result = await platforms.Account.updateAccountHolder({
                        accountHolderCode: accountHolder.accountHolderCode,
                        accountHolderDetails: {
                            ...accountHolderDetails,
                            address: {
                                country: "BE"
                            }
                        }
                    });
                    expect((_a = result.accountHolderDetails.address) === null || _a === void 0 ? void 0 : _a.country).toEqual("BE");
                }
                catch (e) {
                    assertError(e);
                }
            });
            it("should check account holder", async function () {
                nock_1.default.restore();
                try {
                    const result = await platforms.Account.checkAccountHolder({
                        accountHolderCode: accountHolder.accountHolderCode,
                        accountStateType: "Processing",
                        tier: 2
                    });
                    expect(result.resultCode).toEqual("Success");
                }
                catch (e) {
                    assertError(e);
                }
            });
            it("should create an account", async function () {
                nock_1.default.restore();
                try {
                    expect(account.pspReference).toBeDefined();
                }
                catch (e) {
                    assertError(e);
                }
            });
            it("should upload verification document", async function () {
                nock_1.default.restore();
                try {
                    const result = await platforms.Account.uploadDocument({
                        documentContent: documentContent_1.documentContent,
                        documentDetail: {
                            accountHolderCode: account.accountHolderCode,
                            documentType: "ID_CARD_FRONT",
                            description: "test document 000",
                            filename: "IDCardFront.png"
                        }
                    });
                    expect(result.pspReference).toBeDefined();
                }
                catch (e) {
                    assertError(e);
                }
            });
            it("should get uploaded verification documents", async function () {
                nock_1.default.restore();
                try {
                    await platforms.Account.uploadDocument({
                        documentContent: documentContent_1.documentContent,
                        documentDetail: {
                            accountHolderCode: account.accountHolderCode,
                            documentType: "ID_CARD_FRONT",
                            description: "test document 000",
                            filename: "IDCardFront.png"
                        }
                    });
                    const result = await platforms.Account.getUploadedDocuments({
                        accountHolderCode: account.accountHolderCode,
                    });
                    expect(result.documentDetails[0].filename).toEqual("IDCardFront.png");
                }
                catch (e) {
                    assertError(e);
                }
            });
            it("should close account", async function () {
                nock_1.default.restore();
                try {
                    const result = await platforms.Account.closeAccount({
                        accountCode: accountToClose.accountCode
                    });
                    expect(result.status).toEqual("Closed");
                }
                catch (e) {
                    assertError(e);
                }
            });
            it("should suspend account holder", async function () {
                nock_1.default.restore();
                try {
                    const result = await platforms.Account.suspendAccountHolder({
                        accountHolderCode: accountHolderToSuspend.accountHolderCode,
                    });
                    expect(result.pspReference).toBeDefined();
                }
                catch (e) {
                    assertError(e);
                }
            });
            it("should unsuspend account holder", async function () {
                nock_1.default.restore();
                try {
                    const result = await platforms.Account.unSuspendAccountHolder({ accountHolderCode: accountHolderToUnSuspend.accountHolderCode });
                    expect(result.pspReference).toBeDefined();
                }
                catch (e) {
                    assertError(e);
                }
            });
            it("should update account holder state", async function () {
                nock_1.default.restore();
                try {
                    const result = await platforms.Account.updateAccountHolderState({
                        accountHolderCode: accountHolder.accountHolderCode,
                        disable: false,
                        stateType: "Payout"
                    });
                    expect(result.pspReference).toBeDefined();
                }
                catch (e) {
                    assertError(e);
                }
            });
            it("should close account holder", async function () {
                nock_1.default.restore();
                try {
                    const result = await platforms.Account.closeAccountHolder({
                        accountHolderCode: accountHolderToClose.accountHolderCode
                    });
                    expect(result.pspReference).toBeDefined();
                }
                catch (e) {
                    assertError(e);
                }
            });
        });
    });
    describe("Fund", function () {
        it("should retrieve the balance of an account holder", async function () {
            nock_1.default.restore();
            try {
                const result = await platforms.Fund.accountHolderBalance({
                    accountHolderCode: generateRandomCode()
                });
                expect(result.balancePerAccount[0].detailBalance).toBeDefined();
            }
            catch (e) {
                assertError(e);
            }
        });
        it("should retrieve a list of transaction for an account holder's accounts", async function () {
            nock_1.default.restore();
            try {
                const result = await platforms.Fund.accountHolderTransactionList({
                    accountHolderCode: generateRandomCode()
                });
                expect(result.accountTransactionLists[0].transactions).toBeDefined();
            }
            catch (e) {
                assertError(e);
            }
        });
        it("should transfer funds between two accounts", async function () {
            nock_1.default.restore();
            try {
                const result = await platforms.Fund.transferFunds({
                    sourceAccountCode: "8515883280985939",
                    destinationAccountCode: "8815883278206345",
                    amount: {
                        currency: "EUR",
                        value: 1
                    },
                    transferCode: "SUBSCRIPTION"
                });
                expect(result.pspReference).toBeDefined();
            }
            catch (e) {
                assertError(e);
            }
        });
    });
    describe("Notification Configuration", function () {
        let configurationID;
        it("should retrieve all Notification Configurations", async function () {
            nock_1.default.restore();
            try {
                const result = await platforms.NotificationConfiguration.getNotificationConfigurationList({});
                const resultStr = JSON.stringify(result);
                expect(resultStr.includes("pspReference")).toBeTruthy();
            }
            catch (e) {
                assertError(e);
            }
        });
        it("should create a Notification Configuration", async function () {
            nock_1.default.restore();
            try {
                const result = await platforms.NotificationConfiguration.createNotificationConfiguration({
                    configurationDetails: {
                        ...notificationConfigurationDetails,
                        description: `${generateRandomCode()}`
                    }
                });
                expect(result.configurationDetails.active).toBeTruthy();
            }
            catch (e) {
                assertError(e);
            }
        });
        it("should retrieve a Notification Configuration", async function () {
            nock_1.default.restore();
            try {
                configurationID = notificationConfigurationToRetrieve.configurationDetails.notificationId;
                const result = await platforms.NotificationConfiguration.getNotificationConfiguration({
                    notificationId: configurationID
                });
                expect(result.configurationDetails.notifyURL).toEqual("https://www.adyen.com/notification-handler");
            }
            catch (e) {
                assertError(e);
            }
        });
        it("should update a Notification Configuration", async function () {
            nock_1.default.restore();
            try {
                const result = await platforms.NotificationConfiguration.updateNotificationConfiguration({
                    configurationDetails: {
                        eventConfigs: [
                            {
                                eventType: "ACCOUNT_HOLDER_VERIFICATION",
                                includeMode: "EXCLUDE"
                            },
                            {
                                "eventType": "ACCOUNT_CREATED",
                                "includeMode": "INCLUDE"
                            }
                        ],
                        notifyURL: "https://www.adyen.com/notification-handler",
                        notificationId: configurationID
                    }
                });
                const accountHolderVerification = result.configurationDetails.eventConfigs.filter(event => event.eventType === "ACCOUNT_HOLDER_VERIFICATION")[0];
                expect(accountHolderVerification.includeMode).toEqual("EXCLUDE");
            }
            catch (e) {
                assertError(e);
            }
        });
        it("should delete a Notification Configuration", async function () {
            nock_1.default.restore();
            const notificationIds = [];
            notificationIds.push(configurationID);
            try {
                const result = await platforms.NotificationConfiguration.deleteNotificationConfigurations({ notificationIds });
                expect(result.pspReference).toBeDefined();
            }
            catch (e) {
                assertError(e);
            }
        });
    });
});
