// @adyen/api-library/lib/src/security/nexoCrypto.js
"use strict";
/*
 *                       ######
 *                       ######
 * ############    ####( ######  #####. ######  ############   ############
 * #############  #####( ######  #####. ######  #############  #############
 *        ######  #####( ######  #####. ######  #####  ######  #####  ######
 * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
 * ###### ######  #####( ######  #####. ######  #####          #####  ######
 * #############  #############  #############  #############  #####  ######
 *  ############   ############  #############   ############  #####  ######
 *                                      ######
 *                               #############
 *                               ############
 * Adyen NodeJS API Library
 * Copyright (c) 2020 Adyen B.V.
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more info.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const crypto_1 = require("crypto");
const nexoCryptoException_1 = __importDefault(require("../services/exception/nexoCryptoException"));
const invalidSecurityKeyException_1 = __importDefault(require("./exception/invalidSecurityKeyException"));
const nexoDerivedKeyGenerator_1 = __importDefault(require("./nexoDerivedKeyGenerator"));
const nexoConstants_1 = require("../constants/nexoConstants");
var Modes;
(function (Modes) {
    Modes[Modes["ENCRYPT"] = 0] = "ENCRYPT";
    Modes[Modes["DECRYPT"] = 1] = "DECRYPT";
})(Modes || (Modes = {}));
class NexoCrypto {
    static encrypt(messageHeader, saleToPoiMessageJson, securityKey) {
        const derivedKey = nexoDerivedKeyGenerator_1.default.deriveKeyMaterial(securityKey.passphrase);
        const saleToPoiMessageByteArray = Buffer.from(saleToPoiMessageJson, "ascii");
        const ivNonce = NexoCrypto.generateRandomIvNonce();
        const encryptedSaleToPoiMessage = NexoCrypto.crypt(saleToPoiMessageByteArray, derivedKey, ivNonce, Modes.ENCRYPT);
        const encryptedSaleToPoiMessageHmac = NexoCrypto.hmac(saleToPoiMessageByteArray, derivedKey);
        const securityTrailer = {
            adyenCryptoVersion: securityKey.adyenCryptoVersion,
            hmac: encryptedSaleToPoiMessageHmac.toString("base64"),
            keyIdentifier: securityKey.keyIdentifier,
            keyVersion: securityKey.keyVersion,
            nonce: ivNonce.toString("base64"),
        };
        return {
            messageHeader,
            nexoBlob: encryptedSaleToPoiMessage.toString("base64"),
            securityTrailer,
        };
    }
    decrypt(saleToPoiSecureMessage, securityKey) {
        NexoCrypto.validateSecurityKey(securityKey);
        const encryptedSaleToPoiMessageByteArray = Buffer.from(saleToPoiSecureMessage.nexoBlob, "base64");
        const derivedKey = nexoDerivedKeyGenerator_1.default.deriveKeyMaterial(securityKey.passphrase);
        const ivNonce = Buffer.from(saleToPoiSecureMessage.securityTrailer.nonce, "base64");
        const decryptedSaleToPoiMessageByteArray = NexoCrypto.crypt(encryptedSaleToPoiMessageByteArray, derivedKey, ivNonce, Modes.DECRYPT);
        const receivedHmac = Buffer.from(saleToPoiSecureMessage.securityTrailer.hmac, "base64");
        this.validateHmac(receivedHmac, decryptedSaleToPoiMessageByteArray, derivedKey);
        return decryptedSaleToPoiMessageByteArray.toString("ascii");
    }
    static validateSecurityKey(securityKey) {
        const isValid = securityKey
            && securityKey.passphrase
            && securityKey.keyIdentifier
            && !isNaN(securityKey.keyVersion)
            && !isNaN(securityKey.adyenCryptoVersion);
        if (!isValid) {
            throw new invalidSecurityKeyException_1.default("Invalid Security Key");
        }
    }
    static crypt(bytes, dk, ivNonce, mode) {
        const actualIV = Buffer.alloc(nexoConstants_1.NexoEnum.IV_LENGTH);
        for (let i = 0; i < nexoConstants_1.NexoEnum.IV_LENGTH; i++) {
            actualIV[i] = dk.iv[i] ^ ivNonce[i];
        }
        const cipher = mode === Modes.ENCRYPT
            ? crypto_1.createCipheriv("aes-256-cbc", dk.cipherKey, actualIV)
            : crypto_1.createDecipheriv("aes-256-cbc", dk.cipherKey, actualIV);
        let encrypted = cipher.update(bytes);
        encrypted = Buffer.concat([encrypted, cipher.final()]);
        return encrypted;
    }
    static hmac(bytes, derivedKey) {
        const mac = crypto_1.createHmac("sha256", derivedKey.hmacKey);
        return mac.update(bytes).digest();
    }
    static generateRandomIvNonce() {
        return crypto_1.randomBytes(nexoConstants_1.NexoEnum.IV_LENGTH);
    }
    validateHmac(receivedHmac, decryptedMessage, derivedKey) {
        const hmac = NexoCrypto.hmac(decryptedMessage, derivedKey);
        const isValid = hmac.every((item, index) => item === receivedHmac[index]);
        if (!isValid) {
            throw new nexoCryptoException_1.default("Hmac validation failed");
        }
    }
}
exports.default = NexoCrypto;
